core/src/main/java/org/mule/runtime/core/internal/processor/strategy/ComponentProcessor.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.core.internal.processor.strategy;
+
+import org.mule.runtime.core.api.processor.ReactiveProcessor;
+
+/**
+ * Specialization of {@link ReactiveProcessor} that allows to perform certain optimizations when the processor is not blocking, as
+ * indicated by its {@link #isBlocking()} method and that evaluates if a thread switch may happen, as indicated by its
```

Comment:
this sentence is confusing. It's not clear if `mayBeAsync()` should only be considered whtn the operation is blocking, or if the `and` is a typo which should be removed.

Reply:
only isBlocking was left as this is the only aspect from the operation/component which should be taken into account for the optimization.